llvm/test/CodeGen/NVPTX/f32x2-instructions.ll

```diff
@@ -2,16 +2,22 @@
 ; ## Full FP32x2 support enabled by default.
 ; RUN: llc < %s -mcpu=sm_80 -O0 -disable-post-ra -frame-pointer=all            \
 ; RUN: -verify-machineinstrs | FileCheck --check-prefixes=CHECK,CHECK-NOF32X2 %s
-; RUN: %if ptxas-sm_80 %{                                                       \
+; RUN: %if ptxas-sm_80 %{                                                      \
 ; RUN:  llc < %s -mcpu=sm_80 -O0 -disable-post-ra -frame-pointer=all           \
 ; RUN:  -verify-machineinstrs | %ptxas-verify -arch=sm_80                      \
 ; RUN: %}
 ; RUN: llc < %s -mcpu=sm_100 -O0 -disable-post-ra -frame-pointer=all           \
 ; RUN: -verify-machineinstrs | FileCheck --check-prefixes=CHECK,CHECK-F32X2 %s
-; RUN: %if ptxas-sm_100 %{                                                       \
+; RUN: %if ptxas-sm_100 %{                                                     \
 ; RUN:  llc < %s -mcpu=sm_100 -O0 -disable-post-ra -frame-pointer=all          \
 ; RUN:  -verify-machineinstrs | %ptxas-verify -arch=sm_100                     \
 ; RUN: %}
+; RUN: llc < %s -mcpu=sm_100 -O3 -frame-pointer=all -verify-machineinstrs      \
+; RUN:   | FileCheck --check-prefixes=CHECK,CHECK-F32X2-O3 %s
+; RUN: %if ptxas-sm_100 %{                                                     \
+; RUN:  llc < %s -mcpu=sm_100 -O3 -disable-post-ra -frame-pointer=all          \
+; RUN:  -verify-machineinstrs | %ptxas-verify -arch=sm_100                     \
+; RUN: %}
```

Comment:
What's the reason we run with the compilation with `-03` here?

We're generally concerned with the lowering of particular types/instructions. How higher level optimizations affect them is not the job we want/need to do here. In this case, we do add checks that essentially check that we only loaded the used parts of the inputs, but that has little to do with v2f32 we're interested in testing here, IMO.

Reply:
I thought to add it since we still have the issue with loads being lowered inefficiently in debug mode, which you've pointed out.